examples/pytorch/vision/run_image_classification.py

```diff
@@ -0,0 +1,313 @@
+    transformers.utils.logging.enable_explicit_format()
+
+    # Log on each process the small summary:
+    logger.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
+        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
+    )
+    logger.info(f"Training/evaluation parameters {training_args}")
+
+    # Detecting last checkpoint.
+    last_checkpoint = None
+    if os.path.isdir(training_args.output_dir) and training_args.do_train and not training_args.overwrite_output_dir:
+        last_checkpoint = get_last_checkpoint(training_args.output_dir)
+        if last_checkpoint is None and len(os.listdir(training_args.output_dir)) > 0:
+            raise ValueError(
+                f"Output directory ({training_args.output_dir}) already exists and is not empty. "
+                "Use --overwrite_output_dir to overcome."
+            )
+        elif last_checkpoint is not None and training_args.resume_from_checkpoint is None:
+            logger.info(
+                f"Checkpoint detected, resuming training at {last_checkpoint}. To avoid this behavior, change "
+                "the `--output_dir` or add `--overwrite_output_dir` to train from scratch."
+            )
+
+    normalize = T.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
+    train_dataset = ImageFolder(
+        data_args.train_dir,
+        T.Compose(
+            [
+                T.RandomResizedCrop(data_args.image_size),
+                T.RandomHorizontalFlip(),
+                T.ToTensor(),
+                normalize,
+            ]
+        ),
+    )
+    labels = train_dataset.classes
+    label2id, id2label = dict(), dict()
+    for i, label in enumerate(labels):
+        label2id[label] = str(i)
+        id2label[str(i)] = label
+
+    if data_args.validation_dir is not None:
+        eval_dataset = ImageFolder(
+            data_args.validation_dir,
+            T.Compose(
+                [
+                    T.Resize(data_args.image_size),
+                    T.CenterCrop(data_args.image_size),
+                    T.ToTensor(),
+                    normalize,
+                ]
+            ),
+        )
+    else:
+        indices = torch.randperm(len(train_dataset)).tolist()
+        n_val = math.floor(len(indices) * data_args.train_val_split)
+        train_dataset = torch.utils.data.Subset(train_dataset, indices[:-n_val])
+        eval_dataset = torch.utils.data.Subset(train_dataset, indices[-n_val:])
+
+    test_dataset = (
+        ImageFolder(
+            data_args.test_dir,
+            T.Compose(
+                [
+                    T.Resize(data_args.image_size),
+                    T.CenterCrop(data_args.image_size),
+                    T.ToTensor(),
+                    normalize,
+                ]
+            ),
+        )
+        if data_args.test_dir is not None
+        else None
+    )
```

Comment:
Let's reuse the validation transform from before and do an if then else on several lines. There is no point using the one-line syntax `foo = bar if xxx else yyy` when black splits it in several lines.